src/storage/store.js

```diff
@@ -89,7 +89,8 @@ export const SCHEMA = {
 
     preferences: {
       type: "object",
-      additionalProperties: false,
+      // Allow removed preferences to pass validation
+      additionalProperties: true,
```

Comment:
This isn't strictly necisary, we could instead keep around the old preference, but honestly I see little downside to only validating the defined properties and then ignoring the rest. We can eventually write code to clean up orphaned properties.

Reply:
Yeah, I don't remember if there were specific implications we were concerned about that led us to disallowing additional properties. The one thing I can think of is a bit contrived -- this effectively makes the schema mutable as opposed to append-only. Which means we could theoretically delete a property, and then add it back at some later date with a different type, which would break backwards compatibility. There's also a concern of the down-stream use of the stored data. If we allow additional properties, it may open a hole for unexpected data to flow through to other systems. This is a common enough issue that it's listed in security best practices: https://apisecurity.io/encyclopedia/content/oasv3/datavalidation/schema/v3-schema-object-additionalproperties-true.htm
In this particular case it should be safe, since this is scoped to preferences alone, and I don't think we pass the preference structure around.